msrest/async_paging.py

```diff
@@ -0,0 +1,74 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from collections.abc import AsyncIterator
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+class AsyncPagedMixin(AsyncIterator):
+
+    def __init__(self, *args, **kwargs):
+        """Bring async to Paging.
+
+        "async_command" is mandatory keyword argument for this mixin
+        """
+        self._async_get_next = kwargs.get("async_command")
+        if not self._async_get_next:
+            _LOGGER.warning("Paging async iterator protocol is not available for %s",
+                            self.__class__.__name__)
+
+    async def async_get(self, url):
```

Comment:
Is this necessary if you can iterate through items one-by-one now instead of having to check for another page?